eWonConnector-gateway/src/main/java/com/hms_networks/americas/sc/ignition/IgnitionEwonConnectorHook.java

```diff
@@ -0,0 +1,324 @@
+              "Ignition Ewon Connector Module settings have been updated. Restarting module.");
+          restart();
+        }
+      };
+
+  /**
+   * Performs any necessary setup tasks for the Ignition Ewon Connector module.
+   *
+   * @param gatewayContext the current gateway context
+   * @since 1.0.0
+   */
+  @Override
+  public void setup(GatewayContext gatewayContext) {
+    // Set up message
+    LOGGER.debug("Running: Ignition Ewon Connector Module Setup");
+
+    // Store gateway context
+    this.gatewayContext = gatewayContext;
+
+    // Add Ewon Connector to gateway bundles
+    BundleUtil.get()
+        .addBundle(
+            BUNDLE_PREFIX_EWON,
+            IgnitionEwonConnectorHook.class.getClassLoader(),
+            BUNDLE_NAME_EWON_GATEWAY);
+
+    // Set up Ewon Connector settings
+    try {
+      // Create settings tables in internal database, if necessary
+      gatewayContext
+          .getSchemaUpdater()
+          .updatePersistentRecords(EwonConnectorSettings.META, EwonSyncDataState.META);
+
+      // Create Ewon Connector settings panel
+      EwonConnectorSettings settings =
+          gatewayContext.getLocalPersistenceInterface().createNew(EwonConnectorSettings.META);
+
+      // Set settings ID to 0
+      settings.setLong(EwonConnectorSettings.ID, 0L);
+
+      // Ensure created
+      gatewayContext.getSchemaUpdater().ensureRecordExists(settings);
+    } catch (Exception e) {
+      LOGGER.error(
+          "Error validating internal database tables during load of Ignition Ewon Connector module"
+              + " settings.",
+          e);
+    }
+
+    // Set up complete message
+    LOGGER.debug("Finished: Ignition Ewon Connector Module Setup");
+  }
+
+  /**
+   * Performs any necessary startup tasks for the Ignition Ewon Connector module.
+   *
+   * @param licenseState the current license state (not used)
+   * @since 1.0.0
+   */
+  @Override
+  public void startup(LicenseState licenseState) {
+    // Startup message
+    LOGGER.debug("Running: Ignition Ewon Connector Module Startup");
+
+    // Load Ewon Connector settings
+    connectorSettings =
+        gatewayContext.getPersistenceInterface().find(EwonConnectorSettings.META, 0L);
+
+    // Add settings listener
+    EwonConnectorSettings.META.addRecordListener(settingsListener);
+
+    // Verify module is enabled in settings and start up
+    if (connectorSettings.isEnabled()) {
+      // Initialize sync data state manager
+      try {
+        SyncDataStateManager.initialize(gatewayContext);
+      } catch (Exception e) {
+        LOGGER.error(
+            "An error occurred while starting the Ignition Ewon Connector sync data state manager.",
+            e);
+      }
+
+      // Initialize tag manager
+      try {
+        TagManager.initialize(gatewayContext, connectorSettings);
+      } catch (Exception e) {
+        LOGGER.error(
+            "An error occurred while starting the Ignition Ewon Connector tag manager.", e);
```

Comment:
Should the connector continue?  Or should it exit here?

Reply:
Might as well exit -- Adding a startup success flag.